sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_data_lake_directory_client.pyi

```diff
@@ -0,0 +1,23 @@
+# Stubs for azure.storage.filedatalake._data_lake_directory_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._data_lake_file_client import DataLakeFileClient
+from ._models import DirectoryProperties
+from ._path_client import PathClient
+from ._shared.base_client import parse_connection_str
+from typing import Any, Optional
+
+class DataLakeDirectoryClient(PathClient):
+    def __init__(self, account_url: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> DataLakeDirectoryClient: ...
```

Comment:
What does it mean to provide both a connection string **and** a credential object?

Reply:
This is optional if the account URL already has a SAS token, or the connection string already has shared access key values. It can be a token credential class from azure.identity for AAD access.